modules/useQueries.js

```diff
@@ -26,25 +28,40 @@ function useQueries(createHistory) {
       parseQueryString = defaultParseQueryString
 
     function addQuery(location) {
-      if (location.query == null)
-        location.query = parseQueryString(location.search.substring(1))
+      if (location.query == null) {
+        const { search } = location
+        location.query = parseQueryString(search.substring(1))
+        location[SEARCH_BASE_KEY] = { search, searchBase: '' }
```

Comment:
It's for https://github.com/rackt/history/pull/141/files#diff-822f74db4c614a797843213f3d8ad044R71.

Suppose you do `history.push({...location, query: nextQuery})`. Because `useQueries` currently appends the queries to the existing search, this will append the contents of the query to the old search.

I don't expect this to be what anyone wants. The ideal solution here is to change the `useQueries` semantics to overwrite `search` entirely, but I don't want to make a breaking change yet.

This is a bit of a hacky workaround - just check to see what the "intended" search is. It's ugly, but I couldn't figure out anything better that was non-breaking. I think this will work well enough for most practical use cases, and would rather ship this than hold off on a 2.x release for this change.